File: 729-my-calendar-i/my-calendar-i.cpp
class MyCalendar {
private:
    vector<pair<int, int>> l;
public:
    MyCalendar() {
        
    }
    
    bool book(int start, int end) {
        bool flag = true;
        for(auto a : l) {
            if(start < a.second && end > a.first) {
                flag = false;
                break;
            }
        }
        if(flag) {
            l.push_back(make_pair(start, end));
            return true;
        }
        return false;
    }
};

File: 729-my-calendar-i/my-calendar-i.java
class MyCalendar {
    List<List<Integer>> l;
    public MyCalendar() {
        l=new ArrayList<>();
    }
    
    public boolean book(int start, int end) {
        boolean flag = true;
        for(List<Integer> a:l){
            if(start<a.get(1) && end>a.get(0)){
                flag=false;
            }
        }
        if(flag==true){
            List<Integer> a=new ArrayList<>();
            a.add(start);
            a.add(end);
            l.add(new ArrayList<>(a));
            return true;
        }
        return false;
    }
}